mitosheet/src/mito/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -4,17 +4,71 @@ import '../../../../../css/taskpanes/Graph/LoadingSpinner.css';
 import { MitoAPI } from '../../../api/api';
 import { useEffectOnResizeElement } from '../../../hooks/useEffectOnElementResize';
 import useLiveUpdatingParams from '../../../hooks/useLiveUpdatingParams';
-import { AnalysisData, GraphDataArray, GraphParamsBackend, GraphParamsFrontend, GraphSidebarTab, OpenGraphType, SheetData, StepType, UIState } from '../../../types';
+import { AnalysisData, GraphDataArray, GraphParamsBackend, GraphParamsFrontend, GraphSidebarTab, OpenGraphType, RecursivePartial, SheetData, StepType, UIState } from '../../../types';
 import XIcon from '../../icons/XIcon';
 import Col from '../../layout/Col';
 import Row from '../../layout/Row';
 import DefaultEmptyTaskpane from '../DefaultTaskpane/DefaultEmptyTaskpane';
 import { TaskpaneType } from '../taskpanes';
 import GraphSetupTab from './GraphSetupTab';
-import GraphSidebarTabs from './GraphSidebarTabs';
-import GraphStyleTab from './GraphStyleTab';
 import LoadingSpinner from './LoadingSpinner';
 import { convertBackendtoFrontendGraphParams, convertFrontendtoBackendGraphParams, getDefaultGraphParams, getGraphRenderingParams } from './graphUtils';
+import { updateObjectWithPartialObject } from '../../../utils/objects';
+import { classNames } from '../../../utils/classNames';
+import Input from '../../elements/Input';
+
+const Popup = (props: {
+    value: string;
+    xPosition: number;
+    yPosition: number;
+    setValue: (value: string) => void;
+    display: boolean;
+    onClose: () => void;
+    caretPosition?: 'above' | 'below-left' | 'below-centered';
+}) => {
+    if (!props.display) {
+        return <></>
+    }
+    const [ newValue, setNewValue ] = React.useState(props.value);
+    
+    React.useEffect(() => {
+        setNewValue(props.value);
+    }, [props.value]);
+
+    React.useEffect(() => {
+        // Grab focus when we switch the location of the popup
+        const input = document.getElementsByClassName('popup-input')[0] as HTMLInputElement;
+        input.focus();
+    }, [props.xPosition, props.yPosition])
+
+    return (
+        <div
+            className={`graph-element-popup-div popup-div ${props.caretPosition === 'above' ? 'graph-element-popup-div-caret-above' : props.caretPosition === 'below-left' ? 'graph-element-popup-div-caret-below-left' : 'graph-element-popup-div-caret-below-centered'}`}
+            style={{
+                position: 'fixed',
+                left: props.xPosition,
+                top: props.yPosition,
+            }}
+        >
+            <Input
+                className='popup-input'
+                value={newValue}
+                onKeyDown={(e) => {
+                    if (e.key === 'Backspace') {
+                        e.stopPropagation();
+                    }
+                    if (e.key === 'Enter') {
+                        props.setValue(newValue);
+                    }
```

Comment:
Can we rename `props.setValue` to `props.updateGraphTitle` -- and update `setNewValue` to `setEditedTitle`?

Or something to make it more clear that the state in this component is _temporary and could be ditched_ and hasn't been sent to the backend -- while the `setValue` is actually updating the params.